core/domain/activity_jobs_one_off.py

```diff
@@ -246,3 +246,197 @@ def reduce(key, values):
+            reconstituted_rights_model.manager_ids)
+        snapshot_metadata_model.put(update_last_updated_time=False)
+
+    @classmethod
+    def enqueue(cls, job_id, additional_job_params=None):
+        # We can raise the number of shards for this job, since it goes only
+        # over three types of entity class.
+        super(AddMentionedUserIdsContentJob, cls).enqueue(
+            job_id, shard_count=64)
+
+    @classmethod
+    def entity_classes_to_map_over(cls):
+        """Return a list of datastore class references to map over."""
+        return [collection_models.CollectionRightsSnapshotContentModel,
+                exp_models.ExplorationRightsSnapshotContentModel,
+                topic_models.TopicRightsSnapshotContentModel]
+
+    @staticmethod
+    def map(rights_snapshot_model):
+        """Implements the map function for this job."""
+        class_name = rights_snapshot_model.__class__.__name__
+        if isinstance(
+                rights_snapshot_model,
+                collection_models.CollectionRightsSnapshotContentModel):
+            AddMentionedUserIdsContentJob._add_collection_user_ids(
+                rights_snapshot_model)
+        elif isinstance(
+                rights_snapshot_model,
+                exp_models.ExplorationRightsSnapshotContentModel):
+            AddMentionedUserIdsContentJob._add_exploration_user_ids(
+                rights_snapshot_model)
+        elif isinstance(
+                rights_snapshot_model,
+                topic_models.TopicRightsSnapshotContentModel):
+            AddMentionedUserIdsContentJob._add_topic_user_ids(
+                rights_snapshot_model)
+        yield ('SUCCESS-%s' % class_name, rights_snapshot_model.id)
+
+    @staticmethod
+    def reduce(key, ids):
+        """Implements the reduce function for this job."""
+        yield (key, len(ids))
+
+
+class AddMentionedUserIdsMetadataJob(jobs.BaseMapReduceOneOffJobManager):
+    """For every snapshot metadata of a rights model, merge the data from all
+    the user id fields in commit_cmds together and put them in the
+    mentioned_user_ids field of an appropriate RightsSnapshotMetadataModel.
+    """
+
+    @staticmethod
+    def are_commit_cmds_role_change(commit_cmds):
+        """Check if commit_cmds are of a role change type and there is just one
+        commit.
+
+        Args:
+            commit_cmds: list(dict(str, str)). List of commit commands.
+
+        Returns:
+            bool. True when there is one command and it is of role_change type.
+        """
+        role_change_cmds = (
+            feconf.CMD_CHANGE_ROLE,
+            feconf.CMD_REMOVE_MANAGER_ROLE)
+        return (
+            len(commit_cmds) == 1 and commit_cmds[0]['cmd'] in role_change_cmds)
+
+    @staticmethod
+    def _add_col_and_exp_user_ids(snapshot_model):
+        """Merge the user ids from the snapshot and put them in the parent
+        collection rights model.
```

Comment:
I don't see what the implementation of this function has to do with either collections or explorations in particular...